pyatlan/model/assets/asset00.py

```diff
@@ -6372,6 +6424,636 @@ def create(cls, *, asset: Asset, name: str, link: str) -> Link.Attributes:
+        self.attributes.data_domain = data_domain
+
+    @property
+    def output_ports(self) -> Optional[list[Asset]]:
+        return None if self.attributes is None else self.attributes.output_ports
+
+    @output_ports.setter
+    def output_ports(self, output_ports: Optional[list[Asset]]):
+        if self.attributes is None:
+            self.attributes = self.Attributes()
+        self.attributes.output_ports = output_ports
+
+    class Attributes(DataMesh.Attributes):
+        data_product_theme: Optional[str] = Field(
+            None, description="", alias="dataProductTheme"
+        )
+        data_product_status: Optional[DataProductStatus] = Field(
+            None, description="", alias="dataProductStatus"
+        )
+        data_product_criticality: Optional[DataProductCriticality] = Field(
+            None, description="", alias="dataProductCriticality"
+        )
+        data_product_sensitivity: Optional[DataProductSensitivity] = Field(
+            None, description="", alias="dataProductSensitivity"
+        )
+        data_product_assets_d_s_l: Optional[str] = Field(
+            None, description="", alias="dataProductAssetsDSL"
+        )
+        data_product_assets_playbook_filter: Optional[str] = Field(
+            None, description="", alias="dataProductAssetsPlaybookFilter"
+        )
+        data_contracts: Optional[list[DataContract]] = Field(
+            None, description="", alias="dataContracts"
+        )  # relationship
+        data_domain: Optional[DataDomain] = Field(
+            None, description="", alias="dataDomain"
+        )  # relationship
+        output_ports: Optional[list[Asset]] = Field(
+            None, description="", alias="outputPorts"
+        )  # relationship
+
+        @classmethod
+        @init_guid
+        def create(
+            cls,
+            *,
+            name: StrictStr,
+            assets_dsl: StrictStr,
+            icon: Optional[AtlanIcon] = None,
+            domain: Optional[DataDomain] = None,
+            domain_qualified_name: Optional[StrictStr] = None,
+        ) -> DataDomain.Attributes:
+            validate_required_fields(["name"], [name])
+            validate_single_required_field(
+                ["domain", "domain_qualified_name"],
+                [domain, domain_qualified_name],
+            )
+            if domain_qualified_name:
+                domain = DataDomain()
+                domain.unique_attributes = {"qualifiedName": domain_qualified_name}
+            icon_str = icon.value if icon is not None else None
+            camel_case_name = to_camel_case(name)
+            return DataProduct.Attributes(
+                name=name,
+                data_product_assets_d_s_l=assets_dsl,
+                data_domain=domain,
+                mesh_slug=camel_case_name,
+                mesh_abbreviation=camel_case_name,
+                qualified_name=f"default/product/{camel_case_name}",
+                icon=icon_str,
+            )
```

Comment:
As above, I don't see the template that's used to generate this logic in the PR (?)